jax/_src/lax/parallel.py

```diff
@@ -1184,8 +1189,8 @@ def _all_gather_lowering(ctx, x, *, all_gather_dimension, axis_name,
       axis_context,
       (sharding_impls.SPMDAxisContext, sharding_impls.ShardingContext),
   )
-  if (ctx.module_context.platform == 'tpu' or
-      ctx.module_context.platform in ('cuda', 'rocm')
+  if (platform == 'tpu' or
```

Comment:
Nit: `platform in ('tpu', 'cuda', 'rocm')`

Reply:
This would not work because there is an `and` there as well, which applies only to the GPUs.

Reply:
This would be considerably easier to read with parentheses.